pycqed/analysis_v2/timedomain_analysis.py

```diff
@@ -290,17 +400,267 @@ def extract_data(self):
+                    # data), so here we update self.rotation_type to reflect
+                    # this if not already set to some form of PCA.
+                    orig_rot_type = self.rotation_type[qbn]
+                    self.rotation_type[qbn] = 'global_PCA' if \
+                        self.get_param_value('TwoD', default_value=False) \
+                        else 'PCA'
+                    log.warning(f'rotation_type is set to {orig_rot_type} but '
+                                f'no calibration points information was found. '
+                                f'Setting rotation_type for {qbn} to '
+                                f'{self.rotation_type[qbn]}.')
+
+    def get_data_to_fit(self):
+        """
+        Extracts the data_to_fit parameter from the options_dict or metadata
+        and assigns it a reasonable default value based on the cal points
+        information.
+
+        Creates the following attribute:
+            - self.data_to_fit: dict of the form {qbn: proj_data_name} where
+            proj_data_name is a string corresponding to a key in
+            self.proc_data_dict['projected_data_dict'] indicating which
+            projected data the children should fit.
+
+        ! If proj_data_name is a list or tuple of string, this method takes the
+        first entry only !
+        These lists are currently only supported by MultiCZgate_CalibAnalysis
+        and allowing them in this class would break all children.
+        It remains as a to do to upgrade this module in that
+        direction (13.08.2021).
+
+        """
+        # Without deepcopy, the value in the options_dict/metadata will be
+        # modified as well. This will break MultiCZgate_Calib_Analysis which
+        # compares what this class sets for self.data_to_fit and what was given
+        # in options_dict/metadata
+        self.data_to_fit = deepcopy(self.get_param_value('data_to_fit'))
+        if self.data_to_fit is None or not len(self.data_to_fit):
+            # If we have cal points, but data_to_fit is not specified,
+            # choose a reasonable default value. In cases with only two cal
+            # points, this decides which projected plot is generated. (In
+            # cases with three cal points, we will anyways get all three
+            # projected plots.)
+            self.data_to_fit = {}
+            for qbn in self.qb_names:
+                if not len(self.cal_states_dict[qbn]) or \
+                        'pca' in self.rotation_type[qbn].lower():
+                    self.data_to_fit[qbn] = self.rotation_type[qbn]
+                else:
+                    csr = [(k, v) for k, v in
+                           self.cal_states_rotations[qbn].items()]
+                    csr.sort(key=lambda t: t[1])
+                    self.data_to_fit[qbn] = f'p{csr[-1][0]}'
+
+        # TODO: Steph 15.09.2020
+        # This is a hack to allow list inside data_to_fit.
```

Comment:
This is only needed by the MultiCZgate_Calib_Analysis to allow the same data to be fitted in two ways (to extract SWAP errors). We just needed it for this specific case and I didn't want to go change the entire module just for this (it would break all children who assume the values are strings not lists).

Reply:
Yeah, you mentioned something like that, but this explanation should probably go into the comment in the code to avoid that this question comes up again :-)

Reply:
👍

Reply:
Yes, the explanation is there now